client/src/main/java/com/battleships/gamewindow/board/Ship.java

```diff
@@ -17,11 +17,41 @@ public Ship(Set<BoardField> masts) {
         return masts;
     }
 
-    public boolean isMastHit(Coordinate coordinate) {
-        return masts.stream().anyMatch(m -> m.getCoordinate().equals(coordinate));
+    boolean isMastHit(Coordinate coordinate) {
```

Comment:
Change code inside that method to:

        Optional<BoardField> mastForCoordinate = getMastForCoordinate(coordinate);
        return mastForCoordinate.isPresent() ? mastForCoordinate.get().isHit() : false;

Reply:
corrected